File: javascript/algorithms/implementation/cavity-map/cavity-map.js
function cavityMap(grid) {
  for (let i = 1; i < grid.length - 1; i++) {
    for (let j = 1; j < grid[i].length - 1; j++) {
      if (grid[i - 1][j] === `X`) {
        continue;
      }
      if (grid[i][j - 1] < grid[i][j] && grid[i][j + 1] < grid[i][j] && grid[i - 1][j] < grid[i][j] && grid[i + 1][j] < grid[i][j]) {
        grid[i] = `${grid[i].substring(0, j)}X${grid[i].substring(j + 1)}`;
      }
    }
  }
  return grid;
}

module.exports = { cavityMap };

File: python/algorithms/implementation/cavity_map/cavity_map.py
def cavity_map(grid):
  for i in range(1, len(grid) - 1):
    for j in range(1, len(grid[i]) - 1):
      if grid[i][j] == 'X':
        continue
      if (grid[i][j - 1] < grid[i][j] and grid[i][j + 1] < grid[i][j]
          and grid[i - 1][j] < grid[i][j] and grid[i + 1][j] < grid[i][j]):
        grid[i] = grid[i][:j] + 'X' + grid[i][j + 1:]
  return grid